experiment/simulation/js/main.js

```diff
@@ -418,3 +398,62 @@ function observeMessage() {
 function screenWidth() {
   divWidth = document.getElementById("workspace").clientWidth;
 }
+
+async function restart() {
+  // document.getElementById("simulation").style.width = "100%";
+  // document.getElementById("animation-video").style.display = "none";
+  // document.getElementById("plotted-graph-window").style.display = "none";
+  // document.getElementById("head-instructions").innerHTML = "Instructions";
+  // document.getElementById("head-observations").innerHTML = "Instructions";
+  // document.getElementById("instruction").innerHTML = "";
+  // document.getElementById("observation").innerHTML = "";
+  // overallIteration = -3;
+  // iter2 = -1;
+  // iter1 = -1;
+  // setup = 0;
+  // setupMessage();
+  // document.getElementById("experiment-setup").style.display = "block";
+  // document.getElementById("syringe").style.display = "none";
+  // document.getElementById("sample-beaker").style.visibility = "hidden";
+  // document.getElementById("solvent-beaker").style.visibility = "hidden";
+  // document.getElementById("solution-beaker").style.visibility = "hidden";
+  // document.getElementById("spectrometer").style.visibility = "hidden";
+  // document.getElementById("experiment-setup").style.display = "block";
+}
+
+async function observe() {
+  if (overallIteration === 7) {
+    document.getElementById("experiment-setup").style.display = "none";
+    document.getElementById("animation-video").style.display = "block";
+    document.getElementById("animation-bottom-right").play();
+    document.getElementById("head-instructions").innerHTML = "Observations";
+    document.getElementById("head-observations").innerHTML = "Observations";
+    document.getElementById("observation").innerHTML = "";
+    document.getElementById("instruction").innerHTML = "";
+
+    await new Promise((r) => setTimeout(r, 2000));
+    observeMessage();
+    await new Promise((r) => setTimeout(r, 5000));
+    observeMessage();
+    await new Promise((r) => setTimeout(r, 3000));
+    observeMessage();
+    await new Promise((r) => setTimeout(r, 5000));
+    observeMessage();
+    await new Promise((r) => setTimeout(r, 3000));
+    overallIteration++;
```

Comment:
Reading the last 10 lines doesn't give me any idea about what is happening here. Please refactor this code and write it in such a way that the reader gets a high level view of what this code does. Also the repetitive calls to the same function can be better written in the form of a loop which loops over an array of parameters (in this case the timeout)

Reply:
Yes Sir. I will take care of this. However, please ignore this current pull request for the time being. I have made a number of changes to the code according to the functionalities Priya ma'am had suggested in the last meeting. Also, I am testing the code on the hosted website so the final review code may change from the current pull request code.
I will inform you once the code is ready for final review.